src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,112 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.RegistrationException;
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+
+    @Test
+    void register_userNotNull_ok() {
+        User userTest = new User();
+        userTest.setId(123456L);
+        userTest.setLogin("bondorol");
+        userTest.setPassword("123456");
+        userTest.setAge(18);
+        assertNotNull(userTest);
+    }
+
+    @Test
+    void register_userIsNull_notOk() {
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void register_loginIsNull_notOk() {
+        User userTest = new User();
+        userTest.setId(123456L);
+        userTest.setLogin(null);
+        userTest.setPassword("123456");
+        userTest.setAge(18);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userTest);
+        });
+    }
+
+    @Test
+    void register_passwordIsNull_notOk() {
+        User userTest = new User();
+        userTest.setId(123456L);
+        userTest.setLogin("bondorol");
+        userTest.setPassword(null);
+        userTest.setAge(18);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userTest);
+        });
+    }
+
+    @Test
+    void register_ageLessThanMinAge_notOk() {
+        User userTest = new User();
+        userTest.setId(123456L);
+        userTest.setLogin("bondorol");
+        userTest.setPassword("123456");
+        userTest.setAge(17);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userTest);
+        });
+    }
+
+    @Test
+    void register_passwordNull_notOk() {
+        User userTest = new User();
+        userTest.setId(123456L);
+        userTest.setLogin("bondorol");
+        userTest.setPassword(null);
+        userTest.setAge(18);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userTest);
+        });
+    }
+
+    @Test
+    void register_passwordLengthLessThanMin_notOk() {
+        User userTest = new User();
+        userTest.setId(123456L);
+        userTest.setLogin("bondorol");
+        userTest.setPassword("12345");
+        userTest.setAge(18);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userTest);
+        });
+    }
+
+    @Test
+    void register_loginsNotMach_notOk() {
```

Comment:
```suggestion
    void register_loginsAlreadyExist_notOk() {
```
loginsNotMach? What does it mean?